File: app/src/main/java/com/rockspin/bargainbits/di/modules/UrlModule.java
package com.rockspin.bargainbits.di.modules;

import com.rockspin.bargainbits.di.annotations.CheapsharkUrl;

import dagger.Module;
import dagger.Provides;

@Module
public class UrlModule {

    private static final String BASE_CHEAPSHARK_URL = "https://www.cheapshark.com/api/1.0";

    @Provides @CheapsharkUrl public String cheapSharkrUrl(){
        return BASE_CHEAPSHARK_URL;
    }
}

File: app/src/main/kotlin/com/rockspin/bargainbits/di/modules/UrlModule.kt
package com.rockspin.bargainbits.di.modules

import com.rockspin.bargainbits.di.annotations.GameApiUrl
import com.rockspin.bargainbits.di.annotations.GameDealUrl

import dagger.Module
import dagger.Provides

@Module
class UrlModule {

    companion object {
        private val BASE_CHEAPSHARK_URL = "https://www.cheapshark.com"
        private val BASE_CHEAPSHARK_API_URL = "$BASE_CHEAPSHARK_URL/api/1.0"
        private val BASE_CHEAPSHARK_DEAL_URL = "$BASE_CHEAPSHARK_URL/redirect?dealID="
    }

    @Provides @GameApiUrl fun providesGameApiUrl(): String {
        return BASE_CHEAPSHARK_API_URL
    }

    @Provides @GameDealUrl fun providesGameDealUrl(): String {
        return BASE_CHEAPSHARK_DEAL_URL
    }
}
